### DISCUSSION TITLE
Undisclosed Youtube video length limitation or file size download limit?

### DISCUSSION THREAD

**Discussion by heavyharmonies** [aclap-dev/video-downloadhelper#1440] [Help and Q&A]
### Your question

I just purchased a license for the Firefox add-on to be able to download the election coverage streams from Tuesday night. It's hit or miss as to which ones succeed and which ones fail, and I'm wondering if it is due to the length of the video (some are 11+ hours) or the size of the resulting download.

I'm leaning towards the latter, as they all make it to 100% in the rendering process. It is only when the downloads start that they immediately fail.

However, I've been able successfully download streams of 7 hours 46 minutes and 9 hours 49 minutes using the plugin.

This doesn't appear to be a configurable option, and I cannot find any disclosure as to maximum video duration.

Regardless, if there is some sort of limitation, it needs to be disclosed (1) prior to license purchase, and (2) ideally prior to the rendering process starting. There is nothing more annoying than waiting 20+ minutes for a render to complete, have it go to 100% and then immediately fail.

Insanely frustrating.

Thank you.

### Addon & CoApp version and details

version: 9.1.0.44
channel: stable
build date: 2024-09-04
build options: linuxlic: false, noyt: false, browser: firefox
lang: en-US
license: accepted
key (only 16 first characters): a34a8dd26b1d46fd
platform: x86-64 win
UA: Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:132.0) Gecko/20100101 Firefox/132.0
coapp: yes. Info:
{
  "id": "net.downloadhelper.coapp",
  "name": "VdhCoApp",
  "version": "2.0.19",
  "binary": "C:\\Program Files\\DownloadHelper CoApp\\vdhcoapp.exe",
  "displayName": "VdhCoApp",
  "description": "Video DownloadHelper companion app",
  "target": {
    "os": "windows",
    "arch": "x86_64",
    "node": "18"
  },
  "home": "C:\\Users\\Dan",
  "converterBinary": "C:\\Program Files\\DownloadHelper CoApp\\ffmpeg.exe",
  "converterBase": "ffmpeg",
  "converterBaseVersion": "c30f360955"
}

**Comment by rkorns** [comment]
I'm experiencing the same problem.  Most YouTube videos that I download from Firefox work perfectly, however videos that are around an hour or more may truncate the download almost immediately then display 100% downloaded.  I've experimented to see if there is some limiter built in (I'm using the Premium VDH version) and there does not seem to be as some very long videos, as mentioned by heavyharmonies download fine.

The problem does not seem to vary by YouTube channel, some long videos on a channel download completely, others truncate.  I have discovered a glitch where a video I'm watching and decide to download ends up downloading the last previous video I watched with the filename using the title of my current video?  Is this a Firefox caching issue?  If I refresh the page several times, then download, the download and filename are correct --- but sometimes truncate!

Very frustrating.  If it is an extremely important video for me I have to record the stream in real-time!  (Argggghhhh).

Thanks.

RAYK

-----

Platform  Kubuntu 24.04.1 LTS

Firefox
target x86_64-pc-linux-gnu
Compiler 	Version 	Compiler flags
/build/firefox/stage/usr/bin/clang -std=gnu99 	16.0.4 	-pthread -ffunction-sections -fdata-sections -fno-math-errno -fPIC -isystem /build/firefox/parts/firefox/install/usr/include -isystem /build/firefox/stage/usr/include
/build/firefox/stage/usr/bin/clang++ 	16.0.4 	-isystem /build/firefox/parts/firefox/install/usr/include -isystem /build/firefox/stage/usr/include -fno-rtti -pthread -fno-sized-deallocation -fno-aligned-new -ffunction-sections -fdata-sections -fno-math-errno -fno-exceptions -fPIC -isystem /build/firefox/parts/firefox/install/usr/include -isystem /build/firefox/stage/usr/include -gdwarf-4 -O3 -fomit-frame-pointer -funwind-tables
/build/firefox/stage/usr/bin/rustc 	1.76.0 	

VDH Version 9.1.0.44  Last Updated November 4, 2024
